test/spawn.jl

```diff
@@ -428,3 +428,11 @@ if is_unix()
         end
     end
 end
+
+# Test for PR 17803
+let p=Pipe()
+    Base.link_pipe(p; julia_only_read=true, julia_only_write=true)
+    ccall(:jl_static_show, Void, (Ptr{Void}, Any), p.in, "world")
+    @async close(p.in)
+    @test readstring(p.out) == "\"world\""
```

Comment:
The only thing I was wondering about is the cache/blocking behavior of `Pipe` since the test is writing to a pipe without a reader.

This is also not testing what the PR fixes. You should test sth like `Int128(-1)` instead.